test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/StrictCheckSpanProcessor.java

```diff
@@ -27,58 +28,46 @@ public StrictCheckSpanProcessor() {
+        finishedSpanItems.put(span.getSpanId(), toMockSpanData(span));
     }
 
     @Override
     public void onEnd(Span span) {
+        MockSpanData spanData = finishedSpanItems.get(span.getSpanId());
+        if (spanData != null) {
+            spanData.setEndEpochNanos(System.nanoTime());
+            spanData.setHasEnded(true);
+        }
         spanMap.remove(span.getSpanId());
-    }
-
-    /**
-     * Ensures that all the spans are closed. Throws exception message with stack trace of the method form
-     * where the span was created. We can enhance it to print all the failed spans in a single go based on
-     * the usability.
-     */
-    public void ensureAllSpansAreClosed() {
-        if (!spanMap.isEmpty()) {
-            for (Map.Entry<String, StackTraceElement[]> entry : spanMap.entrySet()) {
-                StackTraceElement[] filteredStackTrace = getFilteredStackTrace(entry.getValue());
-                AssertionError error = new AssertionError(
-                    String.format(
-                        Locale.ROOT,
-                        " Total [%d] spans are not ended properly. " + "Find below the stack trace for one of the un-ended span",
-                        spanMap.size()
-                    )
-                );
-                error.setStackTrace(filteredStackTrace);
-                spanMap.clear();
-                throw error;
-            }
+        if (spanMap.containsKey(span.getSpanId())) {
+            spanMap.remove(span.getSpanId());
         }
     }
 
     /**
-     * Clears the state.
+     * Return list of mock span data at any point of time.
      */
-    public void clear() {
-        spanMap.clear();
+    public List<MockSpanData> getFinishedSpanItems() {
+        return new ArrayList<>(finishedSpanItems.values());
     }
 
-    private StackTraceElement[] getFilteredStackTrace(StackTraceElement[] stackTraceElements) {
-        int filteredElementsCount = 0;
-        while (filteredElementsCount < stackTraceElements.length) {
-            String className = stackTraceElements[filteredElementsCount].getClassName();
-            if (className.startsWith("java.lang.Thread")
-                || className.startsWith("org.opensearch.telemetry")
-                || className.startsWith("org.opensearch.tracing")) {
-                filteredElementsCount++;
-            } else {
-                break;
-            }
-        }
-        return Arrays.copyOfRange(stackTraceElements, filteredElementsCount, stackTraceElements.length);
+    private MockSpanData toMockSpanData(Span span) {
+        String parentSpanId = (span.getParentSpan() != null) ? span.getParentSpan().getSpanId() : "";
+        MockSpanData spanData = new MockSpanData(
+            span.getSpanId(),
+            parentSpanId,
+            span.getTraceId(),
+            System.nanoTime(),
+            false,
+            span.getSpanName(),
+            Thread.currentThread().getStackTrace()
+        );
+        ;
```

Comment:
Super nit, please remove `;`
```suggestion
        ;
```